pootle/apps/reports/models/invoice.py

```diff
@@ -0,0 +1,377 @@
+        )
+        missing_required_fields = [
+            field for field in required_fields
+            if field not in config_dict
+        ]
+        if len(missing_required_fields) > 0:
+            raise ImproperlyConfigured(
+                'The configuration for user %s is missing the following required '
+                'fields: %s.\n'
+                'Please double-check your configuration.'
+                % (username, u', '.join(missing_required_fields))
+            )
+
+        return config_dict
+
+    @property
+    def id(self):
+        return self.conf.get('invoice_prefix', '') + self.month_string
+
+    @property
+    def month_string(self):
+        return self.month.strftime(MONTH_FORMAT)
+
+    @lru_cache()
+    def get_rates(self):
+        """Get user rates that were set for the current user during the current
+        month  from start to end. Raise an exception if the user has multiple
+        rates during the period.
+
+        :return: a tuple ``(rate, review_rate, hourly_rate)`` where ``rate`` is
+            the translation rate, and ``review_rate`` is the review rate, and
+            ``hourly_rate`` is the rate for hourly work that can be added as
+            PaidTask.
+        """
+        scores = ScoreLog.objects.for_user_in_range(self.user, self.month_start,
+                                                    self.month_end)
+        rates = scores.values('rate', 'review_rate').distinct()
+        if len(rates) > 1:
+            raise ValueError('Multiple rate values recorded for user %s.' %
+                             (self.user.username))
+
+        rate = rates[0]['rate'] if len(rates) == 1 else 0
+        review_rate = rates[0]['review_rate'] if len(rates) == 1 else 0
+        hourly_rate = 0
+
+        tasks = PaidTask.objects.for_user_in_range(self.user, self.month_start,
+                                                   self.month_end)
+        task_rates = tasks.values('task_type', 'rate').distinct()
+        for task_rate in task_rates:
+            if (task_rate['task_type'] == PaidTaskTypes.TRANSLATION and
+                rate > 0 and
+                task_rate['rate'] != rate):
+                raise ValueError('Multiple TRANSLATION rate values for user %s.'
+                                 % self.user.username)
+            if (task_rate['task_type'] == PaidTaskTypes.REVIEW and
+                review_rate > 0 and
+                task_rate['rate'] != review_rate):
+                raise ValueError('Multiple REVIEW rate values for user %s.' %
+                                 self.user.username)
+            if task_rate['task_type'] == PaidTaskTypes.HOURLY_WORK:
+                if hourly_rate > 0 and task_rate['rate'] != hourly_rate:
+                    raise ValueError('Multiple HOURLY_WORK rate values for user %s.'
+                                     % self.user.username)
+                hourly_rate = task_rate['rate']
+
+        rate = rate if rate > 0 else self.user.rate
```

Comment:
@julen - i thought it worth mentioning - if you are using data in financial calculations - like here:

https://github.com/evernote/zing/blob/master/pootle/apps/accounts/models.py#L93

then you probs want to avoid using FloatFields or similar

> For use cases which require exact decimal representation, try using the decimal module which implements decimal arithmetic suitable for accounting applications and high-precision applications.

https://docs.python.org/3/tutorial/floatingpoint.html
https://docs.python.org/2.7/tutorial/floatingpoint.html